knotx-core/knotx-knot-action/src/main/java/com/cognifide/knotx/knot/action/ActionKnotVerticle.java

```diff
@@ -134,12 +133,18 @@ private void handleFormAction(KnotContext knotContext, Handler<JsonObject> handl
           ClientResponse clientResponse = new ClientResponse(msg.body().getJsonObject("clientResponse"));
           String signal = msg.body().getString("signal");
 
+          if (!HttpResponseStatus.OK.equals(clientResponse.statusCode())) {
```

Comment:
It's easy to miss the exclamation mark at the begining - please make it a small function e.g. ; isNotOkStatus(clientResponse) or something like that..